packages/block-library/src/cover/transforms.js

```diff
@@ -192,6 +195,72 @@ const transforms = {
 					anchor,
 				} ),
 		},
+		{
+			type: 'block',
+			blocks: [ 'core/group' ],
+			isMatch: ( { url } ) => {
+				// If the Cover block uses background media, skip this transform,
+				// and instead use the Group block's default transform.
+				if ( url ) {
+					return false;
+				}
+				return true;
+			},
+			transform: ( attributes, innerBlocks ) => {
+				// Convert Cover overlay colors to comparable Group background colors.
+				const transformedColorAttributes = {
+					backgroundColor: attributes?.overlayColor,
+					gradient: attributes?.gradient,
+					style: attributes?.style
```

Comment:
Do we need to check for `attributes?.style` here? If there is no style, but custom colors in the attributes the custom colors won't migrate. Reported in [this comment.](https://github.com/WordPress/gutenberg/pull/40293#issuecomment-1105836165)

I've been smoke testing the object without this check and seems to function okay so far...

```js
					style: {
						...attributes?.style,
						color: {
							background: attributes?.customOverlayColor,
							gradient: attributes?.customGradient,
							...attributes?.style?.color,
						},
					},
```

Reply:
Thanks will take a look - without the check for the style attribute it was leaving an empty style object serialised in the block attributes - and normally if the style attribute is empty it doesn't get serialised, but that is breaking the custom colors for some reason - will take a look and see if there is a way to not get the empty object, but for custom colors to also work, otherwise we might have to live with the empty attribute.

Reply:
@ramonjd I just pushed a fix for this - keeps custom colors, but still doesn't leave behind an empty style attribute if they aren't set 🤞

Reply:
Thanks @glendaviesnz LGTM